nomt/src/lib.rs

```diff
@@ -421,24 +416,33 @@ impl<T: HashAlgorithm> Nomt<T> {
             .update_and_prove::<T>(compact_actuals, witness);
 
         let mut tx = self.store.new_value_tx();
+
         for (path, read_write) in actuals {
             if let KeyReadWrite::Write(value) | KeyReadWrite::ReadThenWrite(_, value) = read_write {
                 tx.write_value::<T>(path, value);
             }
         }
 
         let merkle_update = merkle_update_handle.join();
+        Ok((tx, merkle_update))
+    }
 
-        let new_root = merkle_update.root;
+    // Effectively commit the transaction.
+    // If 'witness' is set to true, it collects the witness and
+    // returns `(Node, Some(Witness), Some(WitnessedOperations))`
+    // Otherwise, it solely returns the new root node, returning
+    // `(Node, None, None)`
```

Comment:
Looks like this comment was meant to be put on `update_inner` and not `commit_inner`?